src/test/java/org/commcare/formplayer/tests/BaseTestClass.java

```diff
@@ -181,10 +178,11 @@ public void setUp() throws Exception {
         restoreFactoryMock.getSQLiteDB().closeConnection();
         PrototypeUtils.setupThreadLocalPrototypes();
         LocalizerManager.setUseThreadLocalStrategy(true);
+        LocalizerManager.getGlobalLocalizer().addAvailableLocale("en");
+        Localization.setLocale("en");
```

Comment:
@snopoke I needed to set the current locale to fix the `LocaleTextException` I was facing. When running the full test suite, it gets registered/set [here](registerLanguageReference), but when running individual tests it is null. I didn't use this `register` method because I didn't have a valid uri to pass in. Does this seem like the best approach?

Reply:
@shubham1g5 is probably the best person to answer that.

Reply:
Looks harmless to me. Though if it's only specific to this test probably good to localize this inside the `FormEntryTest` itself. If we need to have it in `BaseTestClass`,  I think setting the locale to `default` might be slightly generalized than `en`.   I didn't get how it's getting set when running the full test suite, the link you mentioned seems to be broken. Is it that the locale set as part of a different test is getting retained in this test ?

Reply:
> Is it that the locale set as part of a different test is getting retained in this test ?
Yes
Sorry about that. Meant to link [here](https://github.com/dimagi/formplayer/blob/4a6317119f716b5490772622fe0e1bee1dc8f272/src/main/java/org/commcare/formplayer/engine/FormplayerConfigEngine.java#L179). I am fairly confident that this issue extends to more than just the `FormEntryTest`, though don't have any data to back that up. Simon mentioned he ran into a similar error when running one commcare-core test, but otherwise hasn't had an issue. With that in mind, I'll move into the `FormEntryTest` for now, but if we run into a `LocaleTextException` in another test, this is likely the solution. What is the best way to document that?